RiotSwiftUI/Modules/Room/PollHistory/PollHistoryViewModel.swift

```diff
@@ -104,17 +130,20 @@ private extension PollHistoryViewModel {
 
 extension PollHistoryViewModel.Context {
     var emptyPollsText: String {
-        let days = PollHistoryConstants.chunkSizeInDays
-        
         switch (viewState.bindings.mode, viewState.canLoadMoreContent) {
         case (.active, true):
-            return VectorL10n.pollHistoryNoActivePollPeriodText("\(days)")
+            return VectorL10n.pollHistoryNoActivePollPeriodText("\(syncedPastDays)")
         case (.active, false):
             return VectorL10n.pollHistoryNoActivePollText
         case (.past, true):
-            return VectorL10n.pollHistoryNoPastPollPeriodText("\(days)")
+            return VectorL10n.pollHistoryNoPastPollPeriodText("\(syncedPastDays)")
         case (.past, false):
             return VectorL10n.pollHistoryNoPastPollText
         }
     }
+    
+    var syncedPastDays: UInt {
+        let timeDelta = max(viewState.syncStartDate.timeIntervalSince(viewState.syncedUpTo), 0)
+        return UInt((timeDelta / PollHistoryConstants.oneDayInSeconds).rounded())
```

Comment:
Would it be a tidier implementation to use the calendar in this case? e.g. 
`Calendar.current.dateComponents([.day], from: viewState.syncedUpTo, to: .now)`

Reply:
Applied: https://github.com/vector-im/element-ios/pull/7303/commits/fc8ee97c1b106c20b75cecfebef73c35e036dbec